src/00b3_survey_mics/script.R

```diff
@@ -211,146 +290,183 @@ mics_final <- circ_recoded %>%
     }
   }) %>%
   # remove empty list elements (handy!) and join data
-  purrr::compact() %>% 
-  bind_rows()
+  purrr::compact() %>%
+  bind_rows() %>%
+  select(-mwm6y, -mwm6m, -mwm6d)
+
+
+#' Soft validation: check <25% missing for expected complete variables
+#' This will ensure that variable not coded entirely as NA (unexpectedly
+#' missed during extract)
+
+check_not_na <- function(var, threshold = 0.25) {
+  mics_final %>%
+    group_by(survey_id) %>%
+    summarise(mean(is.na({{var}})) <= threshold) %>%
+    pull() %>%
+    stopifnot()
+}
+
+check_not_na(dob)
+check_not_na(dob)
+check_not_na(area_name)
+check_not_na(circ_status)
+check_not_na(individual_id, 0.0)
+
+#' Handful of missing area_name for TCD2019MICS, mnweight also =0
+mics_final %>%
+  filter(is.na(area_name))
+
+mics_final <- mics_final %>%
+  filter(! (mnweight == 0 & is.na(area_name) & survey_id == "TCD2019MICS"))
+
+check_not_na(area_name, 0.0)
+
+stopifnot(mics_final$circ_status %in% c(0, 1, NA))
 
 # add in remaining columns and fix "missed" recoding in circ_status
-mics_final <- mics_final %>% 
+mics_final <- mics_final %>%
   mutate(
     iso3 = substr(survey_id, 1, 3),
     sex  = "male", # can be assumed
-    # convert CMC -> POSIXct (& Ethiopian to Gregorian Calendar)
-    dob = ifelse(
-      iso3 == "ETH", 
-      cmc_as_date(ethiopian_to_greg(dob)), 
-      cmc_as_date(dob)
-    ),
-    doi = ifelse(
-      iso3 == "ETH", cmc_as_date(ethiopian_to_greg(doi)), cmc_as_date(doi)
-    ),
     # age at interview = floor(date of interview - date of birth)
-    age = floor(lubridate::interval(dob, doi) / lubridate::years(1)),
```

Comment:
I wasn't clear why all of this converting to continuous dates is required. Is it all just to calculate age at interview?

I replaced it with an age calculation based on the CMC dates for interview and birth (which should be consistent with the cleaned reported age at interview in the survey data sets), but let me know if I'm missing the point about the `lubridate` transformations.

Reply:
Yes, it was to calculate age at interview. It was very ad-hoc and stolen from the internet so if you have something more stable or just better that's fine by me!

Reply:
In general, all date calculations with DHS are done with the calendar-month-code (CMC) dates